osf/models/spam.py

```diff
@@ -200,9 +202,12 @@ def do_check_spam(self, author, author_email, content, request_headers):
             'user_agent': request_headers.get('User-Agent'),
             'referer': request_headers.get('Referer'),
         }
-
+        if isinstance(self, Preprint):
+            guid__id = self.get_guid()._id
+        else:
+            guid__id = self.guids.first()._id
         check_resource_for_domains_postcommit(
-            self.guids.first()._id,
+            guid__id,
```

Comment:
If instance is Preprint, maybe we need to pass in the versioned guid id. Can you take a look at `check_resource_for_domains_postcommit` to see how this guid is used? My guess is it probably uses the guid to retrieve the referent.